server/ComputeSuperpixelFeatures/ComputeSuperpixelFeatures.py

```diff
@@ -0,0 +1,387 @@
+
+    print c
+
+    #
+    # Read Input Image
+    #
+    print('\n>> Reading input image ... \n')
+
+    ts = large_image.getTileSource(args.inputImageFile)
+
+    ts_metadata = ts.getMetadata()
+
+    print json.dumps(ts_metadata, indent=2)
+
+    is_wsi = ts_metadata['magnification'] is not None
+
+    #
+    # Compute tissue/foreground mask at low-res for whole slide images
+    #
+    if is_wsi:
+
+        print('\n>> Computing tissue/foreground mask at low-res ...\n')
+
+        im_fgnd_mask_lres, fgnd_seg_scale = \
+            cli_utils.segment_wsi_foreground_at_low_res(ts)
+
+    #
+    # Compute foreground fraction of tiles in parallel using Dask
+    #
+    tile_fgnd_frac_list = [1.0]
+
+    it_kwargs = {
+        'tile_size': {'width': args.analysis_tile_size},
+        'scale': {'magnification': args.analysis_mag},
+    }
+
+    if not process_whole_image:
+
+        it_kwargs['region'] = {
+            'left':   args.analysis_roi[0],
+            'top':    args.analysis_roi[1],
+            'width':  args.analysis_roi[2],
+            'height': args.analysis_roi[3],
+            'units':  'base_pixels'
+        }
+
+    if is_wsi:
+
+        print('\n>> Computing foreground fraction of all tiles ...\n')
+
+        start_time = time.time()
+
+        num_tiles = ts.getSingleTile(**it_kwargs)['iterator_range']['position']
+
+        print 'Number of tiles = %d' % num_tiles
+
+        tile_fgnd_frac_list = htk_utils.compute_tile_foreground_fraction(
+            args.inputImageFile, im_fgnd_mask_lres, fgnd_seg_scale,
+            **it_kwargs
+        )
+
+        num_fgnd_tiles = np.count_nonzero(
+            tile_fgnd_frac_list >= args.min_fgnd_frac)
+
+        percent_fgnd_tiles = 100.0 * num_fgnd_tiles / num_tiles
+
+        fgnd_frac_comp_time = time.time() - start_time
+
+        print 'Number of foreground tiles = %d (%.2f%%)' % (
+            num_fgnd_tiles, percent_fgnd_tiles)
+
+        print 'Time taken = %s' % cli_utils.disp_time_hms(fgnd_frac_comp_time)
+
+    #
+    # Detect and compute nuclei features in parallel using Dask
+    #
+    print('\n>> Detecting superpixels and computing features ...\n')
+
+    start_time = time.time()
+
+    tile_result_list = []
+
+    n_tiles = 0
+
+    for tile in ts.tileIterator(**it_kwargs):
+
+        tile_position = tile['tile_position']['position']
+
+        if is_wsi and tile_fgnd_frac_list[tile_position] <= args.min_fgnd_frac:
+            continue
+
+        n_tiles = n_tiles + 1
+
+        # detect image patches
+        cur_result = dask.delayed(compute_input_tensors)(
+            args.inputImageFile,
+            tile_position,
+            args, **it_kwargs)
+
+        # append result to list
+        tile_result_list.append(cur_result)
+
+    tile_result_list = dask.delayed(tile_result_list).compute()
+
+    print 'Number of tiles processed = %d' % n_tiles
+    print('\n>> tile_result_list to centers and features ...\n')
+
+    slide_centers = np.asarray([centers
```

Comment:
Rename to `superpixel_centers` to be clear.

Reply:
Will be renamed to slide_x_centroids and slide_y_centroids